CONTRIBUTING.md

```diff
@@ -0,0 +1,76 @@
+# Contributing
+
+## Overview
+
+This document explains the processes and practices recommended for contributing enhancements to
+this operator.
+
+- Generally, before developing enhancements to this charm, you should consider [opening an issue
+  ](https://github.com/canonical/training-operator/issues) explaining your use case.
+- If you would like to chat with us about your use-cases or proposed implementation, you can reach
+  us at [Canonical Mattermost public channel](https://chat.charmhub.io/charmhub/channels/charm-dev)
+  or [Discourse](https://discourse.charmhub.io/).
+- Familiarising yourself with the [Charmed Operator Framework](https://juju.is/docs/sdk) library
+  will help you a lot when working on new features or bug fixes.
+- All enhancements require review before being merged. Code review typically examines
+  - code quality
+  - test coverage
+  - user experience for Juju administrators of this charm.
+- Please help us out in ensuring easy to review branches by rebasing your pull request branch onto
+  the `main` branch. This also avoids merge commits and creates a linear Git commit history.
+
+## Developing
+
+You can use the environments created by `tox` for development:
+
+```shell
+tox --notest -e unit
+source .tox/unit/bin/activate
+```
+
+### Testing
+
+```shell
+tox -e lint          # code style
+tox -e unit          # unit tests
+tox -e integration   # integration tests
+tox                  # runs 'lint' and 'unit' environments
+```
+
+## Build charm
+
+Build the charm in this git repository using:
+
+```shell
+charmcraft pack
+```
+
+### Deploy
+
+```bash
+# Create a model
+juju add-model dev
+# Enable DEBUG logging
+juju model-config logging-config="<root>=INFO;unit=DEBUG"
+# Deploy the charm
+juju deploy ./training-operator_ubuntu-20.04-amd64.charm \
+    --resource training-operator-image=kubeflow/training-operator
```

Comment:
Should we add a note here to remind people to add a tag to this image?  Either as a comment, or maybe something like `=kubeflow/training-operator:TAG_IN_MANIFEST`?  Otherwise, who knows whether they're getting the image we expect.  You could also just write here `kubeflow/training-operator:v1-8c6eab2`, but it'll be annoying to need to fix every time we bump versions.

An alternative could be something like: 
```
juju deploy ./training-operator_ubuntu-20.04-amd64.charm \
    --resource oci-image=$(yq eval ".resources.oci-image.upstream-source" metadata.yaml)
```

That way the command is always valid, but no maintenance.  It's kinda non-intuitive to the reader, though

Reply:
Yeah lets update that command to take the image from metadata.